app/models/host/managed.rb

```diff
@@ -209,6 +209,7 @@ def compute_provides?(attr)
   before_validation :set_certname, :if => Proc.new {|h| h.managed? && Setting[:use_uuid_for_certificates] } if SETTINGS[:unattended]
   after_validation :trigger_nic_orchestration, :if => Proc.new { |h| h.managed? && h.changed? }, :on => :update
   before_validation :validate_dns_name_uniqueness
+  validate :validate_owner
```

Comment:
Not exactly, you can have `owner_id.present?`, but type mismatch. Then the `owner` property will remain empty (and will not be set).